src/liboslexec/CMakeLists.txt

```diff
@@ -117,11 +117,48 @@ MACRO ( LLVM_COMPILE llvm_src srclist )
           -Wno-deprecated-register
           -O3 -fno-math-errno -S -emit-llvm -o ${llvm_asm} ${llvm_src}
       COMMAND "${LLVM_DIRECTORY}/bin/llvm-as" -f -o ${llvm_bc} ${llvm_asm}
-      COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/serialize-bc.bash" ${llvm_bc} ${llvm_bc_cpp}
+      COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/serialize-bc.bash" ${llvm_bc} ${llvm_bc_cpp} "osl_llvm_compiled_ops"
       MAIN_DEPENDENCY ${llvm_src}
       DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/serialize-bc.bash"
               ${exec_headers} ${PROJECT_PUBLIC_HEADERS}
       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" )
+
+    # Optionally repeat the bitcode compilation with CUDA-specific options
+    if (CUDA_FOUND)
+      SET ( llvm_asm_cuda "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}_cuda.s" )
+      SET ( llvm_bc_cuda "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}_cuda.bc" )
+      SET ( llvm_bc_cuda_cpp "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}_cuda.bc.cpp" )
+      if (VERBOSE)
+          MESSAGE (STATUS "LLVM_COMPILE CUDA in=${llvm_src_cuda}")
+          MESSAGE (STATUS "LLVM_COMPILE CUDA asm=${llvm_asm_cuda}")
+          MESSAGE (STATUS "LLVM_COMPILE CUDA bc=${llvm_bc_cuda}")
+          MESSAGE (STATUS "LLVM_COMPILE CUDA cpp=${llvm_bc_cuda_cpp}")
+      endif ()
+      SET ( ${srclist} ${${srclist}} ${llvm_bc_cuda_cpp} )
+      ADD_DEFINITIONS (-DOSL_LLVM_CUDA_BITCODE)
+
+      ADD_CUSTOM_COMMAND ( OUTPUT ${llvm_bc_cuda_cpp}
+        COMMAND ${LLVM_BC_GENERATOR}
+            ${LLVM_COMPILE_FLAGS}
+            "-I${CMAKE_CURRENT_SOURCE_DIR}"
+            "-I${CMAKE_SOURCE_DIR}/src/include"
+            "-I${CMAKE_BINARY_DIR}/include"
+            "-I${OPENIMAGEIO_INCLUDE_DIR}"
+            "-I${ILMBASE_INCLUDE_DIR}"
+            "-I${Boost_INCLUDE_DIRS}"
+            -DOSL_COMPILING_TO_BITCODE=1
+            -D__CUDACC__
```

Comment:
I found that here, in the assembly of this command, I also needed:

    --cuda-path="${CUDA_TOOLKIT_ROOT_DIR}"

Reply:
Interesting. I think the lesson here is to take nothing for granted when it comes to build environments. I'll update the commands.